ipmininet/ipnet.py

```diff
@@ -350,6 +350,64 @@ def _broadcast_domains(self):
             domains.append(bd)
         return domains
 
+    def ping6(self, hosts=None, timeout=None):
+        """Ping6 between all specified hosts having at least one IPv6 address
+
+           :param hosts: the list of hosts or None if all must be pinged
+           :param timeout: the time to wait for a response, as string
+           :param returns: the packet loss percentage"""
+        packets = 0
+        lost = 0
+        if not hosts:
+            hosts = self.hosts
+            log.output('*** Ping6: testing ping6 reachability\n')
+        for node in hosts:
+            log.output('%s -> ' % node.name)
+            for dest in hosts:
+                # We do not try if there is no IPv6 address
+                if node != dest and node.intf().ip6:
```

Comment:
`node.intf().ip6 == node.defaultIntf().ip6`

If that particular interface does not have an IPv6 address, others could ...